src/app/InteractionModelEngine.h

```diff
@@ -169,6 +169,12 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
      * Tears down all active subscriptions.
      */
     void ShutdownAllSubscriptions();
+
+    /**
+     * Tears down all subscription handlers.
+     */
+    void ShutdownAllSubscriptionHandlers();
```

Comment:
I don't think this should be in the ReadClient section since it is called by the publisher (ReadHandler).

Reply:
good catch, just fix it, thanks